app/src/main/java/com/example/habithelper/utilities/LinearRegressionCalculator.java

```diff
@@ -1,32 +1,49 @@
 package com.example.habithelper.utilities;
 
 import com.example.habithelper.models.TrackDay;
-import com.parse.ParseUser;
-import org.apache.commons.math3.linear.RealVector;
 import java.util.List;
 
 public class LinearRegressionCalculator {
 
-    private ParseUser currentUser;
     double[] moodList;
     double[][] habitHistoryList;
     public double[] leastSquaresResult;
 
     public LinearRegressionCalculator() {
-        this.currentUser=ParseUser.getCurrentUser();
     }
 
     /**
      * Performs least squares linear regression on the user's list of moods and habits
      *
      * @param daysTracked the list holding every TrackDay Parse object corresponding to the currentUser
+     * @param numDaysTracked the number of days the user has tracked so far
+     * @param numHabits the number of habits the user is tracking
      * @return an array of coefficients corresponding to the correlation of each habit to the user's mood
      */
     public void performLeastSquares(List<TrackDay> daysTracked, int numDaysTracked, int numHabits) {
         moodList = new double[numDaysTracked];
-        habitHistoryList = new double[numDaysTracked][numHabits]; //first is data, second is predictors
+        habitHistoryList = new double[numDaysTracked][numHabits];
 
-        // putting the data from the Parse database into lists that we can perform OLSMultipleLinearRegression on
+        putParseDataIntoLists(daysTracked, numDaysTracked);
+
+        HabitMultipleLinearRegression habitMLR = new HabitMultipleLinearRegression();
+        habitMLR.newSampleData(moodList, habitHistoryList);
```

Comment:
newSampleData can throw an exception, should we catch it and show an error message when it happens?